### ISSUE TITLE
Add documentation for port authors to diagnose potential errors

### ISSUE THREAD

**Issue by donny-dont** [microsoft/vcpkg#3061]
It would be nice to have some additional documentation on potential problems port authors run into while creating their port.

As an example I was working through a port of libressl and ran into https://github.com/libressl-portable/portable/pull/399 while doing so. The problem was they have a generated file that was going into the source directory. It turned out that `vcpkg` was running two configures at once which caused the generated file to get stepped on and turn into garbage and I didn't realize initially that they were running in parallel.

**Comment by JackBoosY** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen this issue.
